lib/uniris/networking/ip_lookup/ip_lookup.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Uniris.Networking.IPLookup do
+  @moduledoc false
+
+  alias __MODULE__.Config
+
+  # Public
+
+  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, binary}
+  def get_node_ip do
+    with {:ok, ip_provider} <- Config.ip_provider() do
```

Comment:
A direct call to the Application configuration could simplify the code with some default value.  Reducing the complexity to read and traverse the code.

Reply:
ok